test/spec/stTable.spec.js

```diff
@@ -380,6 +371,109 @@ describe('st table Controller', function () {
     });
   });
 
+  describe('with stTrimSearchInput set to true', function () {
+    beforeEach(inject(function ($rootScope, $controller, $filter, $parse) {
+      dataSet = [
+        {name: ' Starting with a space'},
+        {name: 'Containing a space'},
+        {name: 'Nospaces'},
+        {name: 'Ending with a space '},
+      ];
+      scope = $rootScope;
+      childScope = scope.$new();
+      scope.data = dataSet;
+      ctrl = $controller('stTableController', {
+        $scope: scope, $parse: $parse, $filter: $filter, $attrs: {
+          stTable: 'data',
+          stTrimSearchInput: 'true'
+        }
+      });
+
+    }));
+
+    describe('search', function () {
+      it('should trim if the input is a string and show all', function () {
+        ctrl.search(' ', 'name');
+        expect(scope.data).toEqual([
+          {name: ' Starting with a space'},
+          {name: 'Containing a space'},
+          {name: 'Nospaces'},
+          {name: 'Ending with a space '},
+        ]);
+      });
+ 
+      it('should trim if the input is a string and search containing h', function () {
+        ctrl.search(' h', 'name');
+        expect(scope.data).toEqual([
+          {name: ' Starting with a space'},
+          {name: 'Ending with a space '},
+        ]);
+      });
+
+      it('should trim if the input is a string and search for everything before a trailing space', function () {
+        ctrl.search('space ', 'name');
+        expect(scope.data).toEqual([
+          {name: ' Starting with a space'},
+          {name: 'Containing a space'},
+          {name: 'Nospaces'},
+          {name: 'Ending with a space '},
+        ]);
+      });
+
+    });
+
+  });    
+
+  describe('with stTrimSearchInput set to false', function () {
+    beforeEach(inject(function ($rootScope, $controller, $filter, $parse) {
+      dataSet = [
+        {name: ' Starting with a space'},
+        {name: 'Containing a space'},
+        {name: 'Nospaces'},
+        {name: 'Ending with a space '},
+    ];
+      scope = $rootScope;
+      childScope = scope.$new();
+      scope.data = dataSet;
+      ctrl = $controller('stTableController', {
+        $scope: scope, $parse: $parse, $filter: $filter, $attrs: {
+          stTable: 'data',
+          stTrimSearchInput: 'false'
+        }
+      });
+
+    }));
+
+    describe('search', function () {
+      it('should NOT trim and show everything containing a space', function () {
+        ctrl.search(' ', 'name');
+        expect(scope.data).toEqual([
+          {name: ' Starting with a space'},
+          {name: 'Containing a space'},
+          {name: 'Ending with a space '}
+        ]);
+      });
+ 
+      it('should NOT trim and search for everything containing the search string including space and s', function () {
+        ctrl.search(' S', 'name');
```

Comment:
Good tests. It captures behaviour on space, leading space, trailing space. 

Consider keeping "s|S" consistent to help those who are not that strong in javascript to not be distracted by the change in casing.